File: backend-spring/src/main/java/com/aalliswell/entity/Vaccination.java
package com.aalliswell.entity;

import com.aalliswell.enums.VaccinationStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "vaccinations", indexes = @Index(name = "idx_vaccinations_user_due", columnList = "user_id,dueDate"))
public class Vaccination extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "baby_id")
    private Baby baby;

    @Column(nullable = false)
    private String vaccineName;

    @Column(nullable = false)
    private LocalDate dueDate;

    private LocalDate completedDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VaccinationStatus status = VaccinationStatus.SCHEDULED;

    private String babyAge;

    @Column(columnDefinition = "TEXT")
    private String notes;
}

File: backend/src/models/Vaccination.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IVaccination extends Document {
    userId: string;
    vaccineName: string;
    dueDate: Date;
    completedDate?: Date;
    status: 'scheduled' | 'completed' | 'missed' | 'rescheduled';
    notes?: string;
    babyAge?: string; // e.g., "6 weeks", "9 months"
}

const VaccinationSchema: Schema = new Schema({
    userId: { type: String, required: true, index: true },
    vaccineName: { type: String, required: true },
    dueDate: { type: Date, required: true },
    completedDate: { type: Date },
    status: {
        type: String,
        enum: ['scheduled', 'completed', 'missed', 'rescheduled'],
        default: 'scheduled'
    },
    notes: { type: String },
    babyAge: { type: String }
}, {
    timestamps: true
});

export default mongoose.model<IVaccination>('Vaccination', VaccinationSchema);
